modules/server/src/mapping/createConnectionResolvers.ts

```diff
@@ -21,38 +23,85 @@ const createConnectionResolvers: CreateConnectionResolversFn = ({
-					  )
-					: type.extendedFields,
-				...(createStateResolvers && {
-					facets: type.config?.[ConfigProperties.FACETS],
-					matchbox: type.config?.[ConfigProperties.MATCHBOX],
-					table: type.config?.[ConfigProperties.TABLE],
-				}),
-			};
+}) => {
+	const configs = async (parentObj, { fieldNames }: { fieldNames: string[] }) => {
+		return {
+			downloads: type.config?.[ConfigProperties.DOWNLOADS],
+			extended: fieldNames
+				? type.extendedFields.filter((extendedField: ExtendedConfigsInterface) =>
+						fieldNames.includes(extendedField.fieldName),
+				  )
+				: type.extendedFields,
+			...(createStateResolvers && {
+				facets: type.config?.[ConfigProperties.FACETS],
+				matchbox: type.config?.[ConfigProperties.MATCHBOX],
+				table: type.config?.[ConfigProperties.TABLE],
+			}),
+		};
+	};
+
+	const aggregationsQuery = resolveAggregations({ type, getServerSideFilter });
+	const aggregationsResolver = async (obj, args, context, info) => {
+		const aggs = await aggregationsQuery(obj, args, context, info);
+		return aggregationsToGraphql(aggs);
+	};
+
+	const defaultHitsResolver = resolveHits({ type, Parallel, getServerSideFilter });
+	const hitsResolver = enableDocumentHits
+		? defaultHitsResolver
+		: resolveHitsFromAggs(aggregationsQuery, dataMaskThreshold);
+
+	return {
+		[type.name]: {
+			aggregations: aggregationsResolver,
+			configs,
+			hits: hitsResolver,
+			// keeping this available for backwards compatibility, but hoping to remove it
+			// TODO: investigate its current usage and need. remove otherwise
+			// Update 2023-02: ENABLE_ADMIN prevents error comes up on facets.
+			// `aggregation` vs numericAggregation` cannot be assessed, requires "mapping".
+			...(enableAdmin && {
+				mapping: async () => {
+					return type.mapping;
+				},
+			}),
 		},
-		...(enableDocumentHits && { hits: resolveHits({ type, Parallel, getServerSideFilter }) }),
-		// keeping this available for backwards compatibility, but hoping to remove it
-		// TODO: investigate its current usage and need. remove otherwise
-		// Update 2023-02: ENABLE_ADMIN prevents error comes up on facets.
-		// `aggregation` vs numericAggregation` cannot be assessed, requires "mapping".
-		...(enableAdmin && {
-			mapping: async () => {
-				return type.mapping;
-			},
-		}),
-	},
-});
+	};
+};
+
+/**
+ * Resolve hits from aggregations
+ * If "aggregations" field is not in query, return 0
+ *
+ * @param aggregationsQuery - resolver ES query code for aggregations
+ * @returns Returns a total count that is less than or equal to the actual total hits in the query.
+ */
+const resolveHitsFromAggs =
+	(aggregationsQuery, dataMaskThreshold) => async (obj, args, context, info) => {
+		/*
+		 * Get "aggregations" field from full query if found
+		 * Popular gql parsing libs parse the "info" property which may not include full query based on schema
+		 */
+		const fileNameConnectionProperty = info.operation.selectionSet.selections[0];
```

Comment:
are all the segments of this chain guaranteed to exist? (including the 0th in the selections)

Reply:
https://github.com/overture-stack/arranger/pull/909/commits/75cb5ab2b25381f200a7798630b92f58713a5d2a
`info` object will always be defined, defaults to empty array